metaserver/src/main/scala/scala/meta/languageserver/compiler/CompletionProvider.scala

```diff
@@ -32,11 +33,28 @@ object CompletionProvider extends LazyLogging {
         isUsedLabel += label
         items += CompletionItem(
           label = label,
-          detail = Some(r.sym.signatureString)
+          detail = Some(r.sym.signatureString),
+          kind = completionItemKind(compiler)(r)
         )
       }
     }
     CompletionList(isIncomplete = false, items = items.result())
   }
 
+  private def completionItemKind(
```

Comment:
If you inline this method to `completions` then you avoid the hairy path dependent signature. scala-compiler has lots of nested methods in inside methods.